### PR TITLE
Sanitize the default task hub names for Azure Storage

### PR DESCRIPTION
**Pull request by ConnorMcMahon**
In a previous release, we started using the site name as the default
task hub value. However, it is possible for the site name not to be a
valid task hub name for Azure Storage, due to having '-' characters and
not meeting the length requirements. We now ensure that no matter the
sitename, the customer will have a valid task hub.

Addresses issue #976

### CODE REVIEW

File: src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs [hunk 1]
```diff
@@ -27,7 +27,15 @@ internal class AzureStorageDurabilityProviderFactory : IDurabilityProviderFactor
             JsonConvert.PopulateObject(JsonConvert.SerializeObject(this.options.StorageProvider), this.azureStorageOptions);
 
             this.azureStorageOptions.Validate();
-            this.azureStorageOptions.ValidateHubName(this.options.HubName);
+            if (this.options.HubName != null // Need to access HubName once to populate IsDefaultHubName
```
Comment (ConnorMcMahon) [inline src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs]: I don't love this. I may need to rework this logic a little bit.

File: src/WebJobs.Extensions.DurableTask/Options/AzureStorageOptions.cs [hunk 2]
```diff
@@ -113,9 +119,37 @@ public void ValidateHubName(string hubName)
...
+
+        internal bool IsSanitizedHubName(string hubName, out string sanitizedHubName)
+        {
+            sanitizedHubName = new string(hubName.ToCharArray()
+                                .Where(char.IsLetterOrDigit)
+                                .Take(MaxTaskHubNameSize)
+                                .ToArray());
+            if (sanitizedHubName.Length < MinTaskHubNameSize)
+            {
+                sanitizedHubName = sanitizedHubName + new string(TaskHubPadding, MinTaskHubNameSize - sanitizedHubName.Length);
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/Options/AzureStorageOptions.cs]: Can you summarize what your strategy is here? I was expecting something that simply stripped dashes and perhaps truncated the name.

```csharp
string siteName = "my-site-1";
string hubName = siteName.Replace("-", "");
hubName = hubName.Substring(0, Math.Min(hubName.Length, 45)) + "Hub";
// result is mysite1Hub
```
I'm a bit confused by the padding logic.

Comment (ConnorMcMahon) [reply src/WebJobs.Extensions.DurableTask/Options/AzureStorageOptions.cs]: The padding is for too short of names (i.e. someone has a site-name like like "ab" or "a-b"). I do like your notion of just appending "Hub" instead of just appending `a`'s to the end.
